File: src/main/java/com/gmail/borlandlp/minigamesdtools/conditions/ConditionsCreatorHub.java
package com.gmail.borlandlp.minigamesdtools.conditions;

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorHub;

public class ConditionsCreatorHub extends CreatorHub {
    public AbstractCondition createCondition(String ID, AbstractDataProvider dataProvider) throws Exception {
        return (AbstractCondition) this.create(ID, dataProvider);
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/conditions/ConditionsCreatorHub.kt
package com.gmail.borlandlp.minigamesdtools.conditions

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.CreatorHub

class ConditionsCreatorHub : CreatorHub() {
    @Throws(Exception::class)
    override fun create(itemID: String, dataProvider: AbstractDataProvider): AbstractCondition {
        return super.create(itemID, dataProvider) as AbstractCondition
    }
}